### PR TITLE
add license-check

### PR DESCRIPTION
**Pull request by Uzlopak**
Can we somehow optimize it with caching?

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/plugins-ci-mysql.yml [hunk 1]
```diff
@@ -30,30 +35,57 @@ jobs:
...
+      - name: Setup node 16
+        uses: actions/setup-node@v3.1.1
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: npm i --ignore-scripts
+
+      - name: Install License-Checker globally
+        run: npm i -g license-checker --ignore-scripts
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci-mysql.yml:61]: Is this needed if we're calling `license-checker` via `npx` anyway?

File: .github/workflows/plugins-ci-mysql.yml [hunk 2]
```diff
@@ -30,30 +35,57 @@ jobs:
...
+    if: >
+      !failure() &&
+      !cancelled() &&
+      inputs.license-check == true
+    name: Check Licenses
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v3.0.2
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci-mysql.yml:48]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/plugins-ci-postgres.yml [hunk 3]
```diff
@@ -30,38 +35,64 @@ jobs:
...
+    if: >
+      !failure() &&
+      !cancelled() &&
+      inputs.license-check == true
+    name: Check Licenses
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v3.0.2
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci-postgres.yml:48]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/plugins-ci.yml [hunk 4]
```diff
@@ -30,34 +35,59 @@ jobs:
...
+    if: >
+      !failure() &&
+      !cancelled() &&
+      inputs.license-check == true
+    name: Check Licenses
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v3.0.2
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci.yml:48]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/plugins-ci.yml [hunk 5]
```diff
@@ -8,6 +8,11 @@ on:
         required: false
         default: 'fastify'
         type: string
+      license-check:
+        description: 'Check Licenses'
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci.yml:12]:
```suggestion
        description: 'Check licenses'
```

File: .github/workflows/plugins-ci-postgres.yml [hunk 6]
```diff
@@ -30,38 +35,64 @@ jobs:
...
+        with:
+          persist-credentials: false
+
+      - name: Setup node 16
+        uses: actions/setup-node@v3.1.1
+        with:
+          node-version: 16
```
Comment (Fdawgs) [inline .github/workflows/plugins-ci-postgres.yml:55]: Probably better to use LTS so we don't have to keep updating the node-version for these actions. :)

```suggestion
      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: lts/*
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by jsumners — APPROVED** [review]
Looks good to me.

(I really need to start my side project around this subject.)

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]
LGTM, will merge and cut a release tomorrow, as there's a few other things I need to tweak in the workflows as well 😄

**Pull request merged**